importlib-metadata: pytester test failure

`testing/test_pytester.py::TestInlineRunModulesCleanup::test_inline_run_taking_and_restoring_a_sys_modules_snapshot` fails for me due to https://github.com/pytest-dev/pytest/commit/13f02af97d676bdf7143aa1834c8898fbf753d87:

```
        assert spy._spy_restore_count == 1
>       assert sys.modules == original, [
            [x for x in sys.modules if x not in original],
            [x for x in original if x not in sys.modules],
        ]
E       AssertionError: [['pkg_resources', 'plistlib', 'xml.parsers', 'xml.parsers.expat', 'pyexpat.errors', 'pyexpat.model', ...], []]
E       assert {'__future__': <module '__future__' from '/usr/lib/python3.7/__future__.py'>, '__main__': <module 'pytest' from '…/Vcs/pytest/src/pytest.py'>, '_abc': <module '_abc' (built-in)>, '_ast': <module '_ast' (built-in)>, ...} == {'__future__': <module '__future__' from '/usr/lib/python3.7/__future__.py'>, '__main__': <module 'pytest' from '…/Vcs/pytest/src/pytest.py'>, '_abc': <module '_abc' (built-in)>, '_ast': <module '_ast' (built-in)>, ...}
E        +  where {'__future__': <module '__future__' from '/usr/lib/python3.7/__future__.py'>, '__main__': <module 'pytest' from '…/Vcs/pytest/src/pytest.py'>, '_abc': <module '_abc' (built-in)>, '_ast': <module '_ast' (built-in)>, ...} = sys.modules

…/Vcs/pytest/testing/test_pytester.py:247: AssertionError
---------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------
============================= test session starts ==============================
platform linux -- Python 3.7.3, pytest-4.6.1.dev186+gc9923a3a5, py-1.8.0, pluggy-0.12.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('…/Vcs/pytest/.hypothesis/examples')
rootdir: /tmp/pytest-of-user/pytest-37/test_inline_run_taking_and_restoring_a_sys_modules_snapshot0
plugins: hypothesis-4.23.8
collected 1 item

test_inline_run_taking_and_restoring_a_sys_modules_snapshot.py .         [100%]

=========================== 1 passed in 0.01 seconds ===========================
========================================================================== short test summary info ===========================================================================
FAILED testing/test_pytester.py::TestInlineRunModulesCleanup::test_inline_run_taking_and_restoring_a_sys_modules_snapshot - AssertionError: [['pkg_resources', 'plistlib', ...

```

(I've changed the assertion there to better see the difference)

Code: https://github.com/pytest-dev/pytest/blob/157515f3c5b132eb1c126003c6d31a9bf36b1059/testing/test_pytester.py#L229-L248
Also fails in tox (py37).

/cc @asottile

--- a/testing/test_pytester.py
+++ b/testing/test_pytester.py
@@ -231,8 +231,8 @@ class TestInlineRunModulesCleanup:
     ):
         spy_factory = self.spy_factory()
         monkeypatch.setattr(pytester, "SysModulesSnapshot", spy_factory)
-        original = dict(sys.modules)
         testdir.syspathinsert()
+        original = dict(sys.modules)
         testdir.makepyfile(import1="# you son of a silly person")
         testdir.makepyfile(import2="# my hovercraft is full of eels")
         test_mod = testdir.makepyfile(

